### ISSUE TITLE
False positive: Using tuple struct with actix_web::web::Data

### ISSUE THREAD

**Issue by LauH1987** [intellij-rust/intellij-rust#9852]
## Environment

* **IntelliJ Rust plugin version:**
0.4.184.5057-223
* **Rust toolchain version:**
rustc 1.64.0
cargo 1.64.0
* **IDE name and version:**
IntelliJ IDEA 2022.3 (Community Edition)
Build #IC-223.7571.182
* **Operating system:**
WSL2
## Problem description
When using a tuple struct with Actix Web's `Data` type, it fails to understand the deref of `&foo.0` to the inner type.
The code compiles and runs with `cargo run`, but IntelliJ shows the error:

> mismatched types [E0308] expected `&str`, found `&Arc<TupleStruct>`

## Steps to reproduce
`cargo.toml`
```toml
[package]
name = "sample"
version = "0.1.0"
edition = "2021"

[dependencies]
actix-web = "4"
```

`main.rs`
```Rust
use actix_web::web::Data;

fn main() {
    let tuple_struct = TupleStruct("Hello".to_string());
    let data = Data::new(tuple_struct);
    caller(data);
}

struct TupleStruct(String);

fn caller(data: Data<TupleStruct>,){
   print_str(&data.0);
}
fn print_str(input: &str) {
    println!("{input}");
}
```
![image](https://user-images.githubusercontent.com/22587888/205894275-912daed1-995e-449b-be83-2dd78432d2df.png)
